libs/libdaemon/src/BaseDaemon.cpp

```diff
@@ -804,7 +806,8 @@ void BaseDaemon::buildLoggers(Poco::Util::AbstractConfiguration & config)
         error_log_file->setProperty(Poco::FileChannel::PROP_PURGECOUNT, config.getRawString("logger.count", "10"));
         error_log_file->setProperty(Poco::FileChannel::PROP_FLUSH, config.getRawString("logger.flush", "true"));
         error_log_file->setProperty(Poco::FileChannel::PROP_ROTATEONOPEN, config.getRawString("logger.rotateOnOpen", "false"));
-        errorlog->setChannel(error_log_file);
+        error_log_file_async = new Poco::AsyncChannel(error_log_file);
```

Comment:
`AsyncChannel` will create a thread. It means that each time the `tiflash.toml` is modified, a thread will be created.
The previously created thread in `AsyncChannel` will still exist because the old `log` is not released.

Reply:
I add
```cpp
if (error_log_file_async)
    error_log_file_async->close();
```
before `error_log_file_async = new Poco::AsyncChannel(error_log_file);`. Is it able to solve this problem?

Reply:
I add
```cpp
if (error_log_file_async)
    error_log_file_async->close();
```
before `error_log_file_async = new Poco::AsyncChannel(error_log_file);`. Is it able to solve this problem?

Reply:
> I add
> 
> ```c++
> if (error_log_file_async)
>     error_log_file_async->close();
> ```
> 
> before `error_log_file_async = new Poco::AsyncChannel(error_log_file);`. Is it able to solve this problem?

1. The `logger` that use `error_log_file_async` may not be closed, so the `error_log_file_async` in these `logger` should not be closed.
2. `AsyncChannel::close` will wait until the `_queue` is empty, which may stuck the thread. 

Can we `setChannel` with the old `error_log_file_async` without `close`?

Reply:
in this way?
```cpp
error_log_file->setProperty(Poco::FileChannel::PROP_ROTATEONOPEN, config.getRawString("logger.rotateOnOpen", "false"));
if (!error_log_file_async)
      error_log_file_async = new Poco::AsyncChannel(error_log_file);
errorlog->setChannel(error_log_file_async);
```

Reply:
> in this way?
> 
> ```c++
> error_log_file->setProperty(Poco::FileChannel::PROP_ROTATEONOPEN, config.getRawString("logger.rotateOnOpen", "false"));
> if (!error_log_file_async)
>       error_log_file_async = new Poco::AsyncChannel(error_log_file);
> errorlog->setChannel(error_log_file_async);
> ```

LGTM